annotation-file-utilities/src/annotator/find/InheritedSymbolFinder.java

```diff
@@ -0,0 +1,211 @@
+ */
+public class InheritedSymbolFinder {
+  private static Source source = null;
+
+  // class (flat) name -> method index
+  private static Map<String, Multimap<String, ExecutableElement>> cache =
+      new TreeMap<String, Multimap<String, ExecutableElement>>();
+
+  // keys: class (flat) name
+  // values: multimap of method (simple) name to matching method elems
+  private static Map<String, Multimap<String, ExecutableElement>> cash =
+      new HashMap<String, Multimap<String, ExecutableElement>>();
+
+  // initialized once and for all
+  private static Multimap<String, ExecutableElement> objectMethods = null;
+
+  private InheritedSymbolFinder() {}
+
+  // Needed for obtaining instances of utility classes Types and Elements.
+  // TODO: make methods non-static and pass source to constructor instead.
+  public static void setSource(Source source) {
+    if (InheritedSymbolFinder.source != source) {
+      InheritedSymbolFinder.source = source;
+      cache.clear();
+      cash.clear();
+      if (source != null) {
+        // initialize objectMethods
+        if (objectMethods == null) {
+          TypeElement o =
+              source.getElementUtils().getTypeElement("java.lang.Object");
+          objectMethods =
+              LinkedHashMultimap.<String, ExecutableElement>create();
+          for (Element e : o.getEnclosedElements()) {
+            if (e.getKind() == ElementKind.METHOD) {
+              String key = keyFor((Symbol.MethodSymbol) e);
+              objectMethods.put(key, (ExecutableElement) e);
+            }
+          }
+        }
+        cash.put("java.lang.Object", objectMethods);
+      }
+    }
+  }
+
+  /**
+   * Finds whether a method matching a JVML descriptor has an inherited
+   *  implementation in a given class.
+   *
+   * @param classSymbol symbol table entry for a class
+   * @param methodDescriptor JVML descriptor for a method
+   * @return whether a matching method exists in the class
+   */
+  public static boolean isInheritedIn(Symbol.ClassSymbol classSymbol,
+      String methodDescriptor) {
+    Multimap<String, ExecutableElement> inherited = getInherited(classSymbol);
+    int ix = methodDescriptor.indexOf('(');
+    String key = methodDescriptor.substring(0, ix);
+    Collection<ExecutableElement> elems = inherited.get(key);
+
+    // There may be multiple methods stored under the same key (bare
+    // method name), so check each of them.
+    // TODO: figure out good way to determine whether method descriptor
+    // gives the type of an inherited method.  (Only exact matches work
+    // now.)
+    if (elems != null) {
+      for (ExecutableElement elem : elems) {
+        String jvmName = JVMNames.getJVMMethodName(elem);
+        int ix0 = methodDescriptor.lastIndexOf(')') + 1;
+        int ix1 = jvmName.lastIndexOf(')') + 1;
+        String key0 = methodDescriptor.substring(0, ix0);
+        String key1 = jvmName.substring(0, ix1);
+        if (key0.equals(key1)) { return true; }
```

Comment:
Although it takes a bit more vertical space, it's standard Java style to put the then body on a separate line even if it's only one line long.  Please do so, here and elsewhere.